test/widgets/theme_test.dart

```diff
@@ -0,0 +1,78 @@
+import 'package:checks/checks.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:zulip/widgets/text.dart';
+import 'package:zulip/widgets/theme.dart';
+
+import '../flutter_checks.dart';
+
+void main() {
+  group('button text size and letter spacing', () {
+    Future<void> doCheck(
+      String description, {
+      required Widget Function(BuildContext context, String text) buttonBuilder,
+      double? ambientTextScaleFactor,
+    }) async {
+      testWidgets(description, (WidgetTester tester) async {
+        if (ambientTextScaleFactor != null) {
+          tester.platformDispatcher.textScaleFactorTestValue = ambientTextScaleFactor;
+          addTearDown(tester.platformDispatcher.clearTextScaleFactorTestValue);
+        }
+        const buttonText = 'Zulip';
+        double? expectedFontSize;
+        double? expectedLetterSpacing;
+        await tester.pumpWidget(
+          Builder(builder: (context) => MaterialApp(
+            theme: zulipThemeData(context),
+            home: Builder(builder: (context) {
+              expectedFontSize = Theme.of(context).textTheme.labelLarge!.fontSize!;
+              expectedLetterSpacing = proportionalLetterSpacing(context,
+                0.01, baseFontSize: expectedFontSize!);
+              return Builder(builder: (context) =>
+                buttonBuilder(context, buttonText));
+            }))));
+
+        final text = tester.renderObject<RenderParagraph>(find.text(buttonText)).text;
+        check(text.style!)
+          ..fontSize.equals(expectedFontSize)
+          ..letterSpacing.equals(expectedLetterSpacing);
+      });
+    }
+
+    doCheck('with device text size adjusted',
+      ambientTextScaleFactor: 2.0,
+      buttonBuilder: (context, text) => ElevatedButton(onPressed: () {},
+        child: Text(text)));
```

Comment:
It seems like none of these builders actually use the context. So they can be simplified to just widgets (by pulling up the `buttonText` constant too):
```suggestion
      button: ElevatedButton(onPressed: () {}, child: const Text(buttonText)));
```

Reply:
Ah right, cool!